runtime/services/catalog/artifacts/artifacts.go

```diff
@@ -43,7 +45,23 @@ func Read(ctx context.Context, repoStore drivers.RepoStore, instID, filePath str
 		return nil, ErrFileRead
 	}
 
-	catalog, err := artifact.DeSerialise(ctx, filePath, blob)
+	instance, err := registryStore.FindInstance(ctx, instID)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert templatised artifact
+	t, err := template.New("source").Option("missingkey=error").Parse(blob)
+	if err != nil {
+		return nil, err
+	}
+
+	bw := new(bytes.Buffer)
+	if err := t.Execute(bw, instance.Env); err != nil {
+		return nil, err
+	}
```

Comment:
The `map[string]map[string]string` providing `.env.KEY` should be crafted here

Reply:
It will also need to merge `inst.Env` and `inst.DefaultEnv` (can add a util function on `drivers.Instance` for that)